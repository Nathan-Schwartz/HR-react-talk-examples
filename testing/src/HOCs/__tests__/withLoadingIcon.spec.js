import React from 'react';
import { mount } from 'enzyme';
import { spy } from 'sinon';
import toJSON from 'enzyme-to-json';

import withLoadingIcon from '../withLoadingIcon';


const mockLoadingState = {
  loading: true,
  data: null,
}

const mockLoadedState = {
  loading: false,
  data: { present: true },
}

const BasicComponent = (props) => <span>{JSON.stringify(props.data)}</span>

describe('', () => {
  let mountedComponent;
  let ComposedComponent;

  it('should return a Component', () => {
    ComposedComponent = withLoadingIcon(BasicComponent);
    React.isValidElement(ComposedComponent);
  })

  it('should mount', () => {
    mountedComponent = mount(
      <ComposedComponent {...mockLoadedState} />,
    );

    expect(mountedComponent).toBeTruthy();
  });

  it('Should match loaded snapshot', () => {
    expect(toJSON(mountedComponent)).toMatchSnapshot();
  });

  it('Should match loading snapshot', () => {
    mountedComponent.setProps(mockLoadingState);
    expect(toJSON(mountedComponent)).toMatchSnapshot();
  });
});