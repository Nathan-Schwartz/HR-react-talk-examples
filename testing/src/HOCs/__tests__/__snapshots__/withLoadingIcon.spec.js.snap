// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match loaded snapshot 1`] = `
<ConditionalLoader
  data={
    Object {
      "present": true,
    }
  }
  loading={false}
>
  <BasicComponent
    data={
      Object {
        "present": true,
      }
    }
    loading={false}
  >
    <span>
      {"present":true}
    </span>
  </BasicComponent>
</ConditionalLoader>
`;

exports[`Should match loading snapshot 1`] = `
<ConditionalLoader
  data={null}
  loading={true}
>
  <img
    src="https://media.giphy.com/media/3oEjI6SIIHBdRxXI40/giphy.gif"
    style={
      Object {
        "height": 100,
        "margin": "0 auto",
        "width": 100,
      }
    }
  />
</ConditionalLoader>
`;
